merlin/dataloader/tensorflow.py

```diff
@@ -305,6 +311,35 @@ def _cast_to_numpy_dtype(self, dtype):
         Get the numpy dtype from the framework dtype.
         """
         return dtype.as_numpy_dtype()
+    
+    def _reshape_dim(self, tensor):
+        if self.tensors_as_1d:
+            return tf.reshape(tensor, shape=[-1])
```

Comment:
👍🏻 

In the course of researching this change, I learned that `reshape` is essentially free because it just changes how the data in memory is interpreted, but `transpose` actually copies the data so it's more expensive. In theory, either one would work here, but `reshape` is definitely the better choice.